pkg/apis/pipeline/v1/result_validation.go

```diff
@@ -51,3 +53,26 @@ func (tr TaskResult) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 	return nil
 }
+
+// validateObjectResult validates the object result and check if the Properties is missing
+// for Properties values it will check if the type is string.
```

Comment:
```suggestion
// validateObjectResult validates that the object result has Properties,
// and that all Properties are strings.
```

Reply:
Also, is there a reason PropertySpec has a type if the only supported type is string?

Reply:
yeah it is required in [tep75](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#use-cases) and I think the reason is to support future extension